query/src/main/java/io/keen/client/java/PersistentAnalysis.java

```diff
@@ -0,0 +1,114 @@
+package io.keen.client.java;
+
+import io.keen.client.java.http.HttpMethods;
+
+/**
+ * Base class that represents requests to the endpoints for managing persistent analyses, such as
+ * <a href="https://keen.io/docs/api/#saved-queries">Saved/Cached Queries</a> and
+ * <a href="https://keen.io/docs/api/#cached-datasets">Cached Datasets</a>.
+ *
+ * Should be an interface, but KeenQueryRequest is meant to be a package-private interface defining
+ * package-private functionality, so until the next major version when we fix up public interface
+ * surface area, it's a package-private abstract class.
+ *
+ * @author masojus
+ */
+abstract class PersistentAnalysis extends KeenQueryRequest {
+    // TODO : Does the API support unicode non-ASCII resource names? HTTP URIs can't technically
+    // contain non-ASCII, but browsers can choose to display URL-encoded characters and choose an
+    // encoding to use from the page itself, but at least via the Explorer UI, giving a query a
+    // display name of "Can has diacritics Ñüáúéíó üöäñ çàèìòù ãõ" yields a url as follows:
+    // "https://...?saved_query=can-has-diacritics----". So two questions remain:
+    // 1) Can the API handle a queryName with non-ASCII? Should we allow ?
+    // 2) This regex does not match strings like "árbol π" so we only accept ASCII
```

Comment:
Ok that's what it does. I'll remove the comment or adjust it to reflect what and why.